doc/cifarSC2011/advanced_theano.txt

```diff
@@ -179,96 +180,200 @@ Inplace optimization
+        output 0: dtype=int64, shape=(), strides=c
+       ... (remaining 1 Apply instances account for 0.10%(0.00s) of the runtime)
+
+    Memory Profile
+    (Sparse variables are ignored)
+    (For values in brackets, it's for linker = c|py
+    ---
+        Max if no gc (allow_gc=False): 2469KB (2469KB)
+        CPU: 1242KB (1242KB)
+        GPU: 1227KB (1227KB)
+    ---
+        Max if linker=cvm(default): 2466KB (2464KB)
+        CPU: 1241KB (1238KB)
+        GPU: 1225KB (1227KB)
+    ---
+        Memory saved if views are used: 1225KB (1225KB)
+        Memory saved if inplace ops are used: 17KB (17KB)
+        Memory saved if gc is enabled: 3KB (4KB)
+    ---
+
+        <Sum apply outputs (bytes)> <Apply outputs shape> <created/inplace/view> <Apply node>
+
+           1254400B  [(400, 784)] c GpuFromHost(x)
+           1254400B  [(784, 400)] v GpuDimShuffle{1,0}(GpuFromHost.0)
+           1254400B  [(784, 400)] c HostFromGpu(GpuDimShuffle{1,0}.0)
+              6272B  [(784,)] c dot(x.T, Elemwise{Composite{(((scalar_sigmoid(i0) * i1 * i2) / i3) - ((i4 * i1 * i5) / i3))}}[(0, 0)].0)
+              6272B  [(784,)] i Elemwise{Composite{(i0 - (i1 * (i2 + (i3 * i0))))}}[(0, 0)](w, TensorConstant{(1,) of 0...0000000149}, dot.0, TensorConstant{(1,) of 0...9999999553})
+              3200B  [(400,)] c dot(x, w)
+              3200B  [(400,)] i Elemwise{Composite{((-i0) - i1)}}[(0, 0)](dot.0, InplaceDimShuffle{x}.0)
+              3200B  [(400,)] i Elemwise{ScalarSigmoid}[(0, 0)](Elemwise{neg,no_inplace}.0)
+              3200B  [(400,)] c Elemwise{neg,no_inplace}(Elemwise{Composite{((-i0) - i1)}}[(0, 0)].0)
+              3200B  [(400,)] i Elemwise{Composite{(((scalar_sigmoid(i0) * i1 * i2) / i3) - ((i4 * i1 * i5) / i3))}}[(0, 0)](Elemwise{Composite{((-i0) - i1)}}[(0, 0)].0, TensorConstant{(1,) of -1.0}, y, Elemwise{Cast{float64}}.0, Elemwise{ScalarSigmoid}[(0, 0)].0, HostFromGpu.0)
+              3200B  [(400,)] c Elemwise{Composite{((i0 * scalar_softplus(i1)) - (i2 * i3 * scalar_softplus(i4)))}}(y, Elemwise{Composite{((-i0) - i1)}}[(0, 0)].0, TensorConstant{(1,) of -1.0}, HostFromGpu.0, Elemwise{neg,no_inplace}.0)
+              1600B  [(400,)] i GpuElemwise{Sub}[(0, 1)](CudaNdarrayConstant{[ 1.]}, GpuFromHost.0)
+              1600B  [(400,)] c HostFromGpu(GpuElemwise{Sub}[(0, 1)].0)
+              1600B  [(400,)] c GpuFromHost(y)
+       ... (remaining 7 Apply account for  448B/3800192B ((0.01%)) of the Apply with dense outputs sizes)
+
+        <created/inplace/view> is taken from the Op's declaration.
+        Apply nodes marked 'inplace' or 'view' may actually allocate memory, this is not reported here. If you use DebugMode, warnings will be emitted in those cases.
+
+    Here are tips to potentially make your code run faster
+                     (if you think of new ones, suggest them on the mailing list).
+                     Test them first, as they are not guaranteed to always provide a speedup.
+      Sorry, no tip for today.
```

Comment:
I think we should investigate that. This example keeps a `dot` in the graph, because of mixed `float32`/`float64` dtype, and that is definitely something that should be warned about in the tips, I thought we did check for that.